cherab/core/math/function/vectorfunction3d.pyx

```diff
@@ -49,13 +49,38 @@ cdef class VectorFunction3D:
         return self.evaluate(x, y, z)
 
 
+cdef class ConstantVector3D(VectorFunction3D):
```

Comment:
Do you mean there should be separate `constantvector2d.pyx` and `constantvector3d.pyx` files, and associated `pxd` files? How far do we take this? Do we reproduce the same file structure as raysect has, by splitting 2D and 3D into separate directories? Seems like overkill for so few classes, unless we want to greatly expand the vector function framework in a similar way to raysect's function framework.

Reply:
Essentially yes, the same as Raysect. Might seem overkill now, but it'll make it cleaner to add things later.

Reply:
@CnlPepper why don't we just move all of the Vector Function framework to Raysect too? There's nothing Cherab-specific in this: it's based on the Raysect `Vector3D` class. I don't want to spend time rearranging this to match the Raysect framework and then we end up having to reimplement it all in Raysect later on.

We could put some suitable imports in `cherab/core/math/__init__.py` to avoid breaking downstream codes, similar to how `Function2D` and `Function3D` from Raysect are imported into the `cherab.core.math` namespace now.

Reply:
If you wish to re-implement it in Raysect you can, but for licensing reasons we cannot just transfer the code. Raysect has a more liberal license.

Reply:
I will probably, like we did with the scalar function framework, only implement the minimal required set initially. Further enhancements can be raised in separate issues as and when they're required. This'll mean we can release it sooner.

Reply:
This is now implemented fully in raysect 0.7.